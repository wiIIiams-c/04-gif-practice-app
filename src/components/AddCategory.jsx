import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onFormSubmit = (event) => {
        event.preventDefault()
        if(inputValue.trim().length <= 1) return;

        // setCategories(categories => [inputValue, ...categories])
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return(
        <form onSubmit={onFormSubmit}>
            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}