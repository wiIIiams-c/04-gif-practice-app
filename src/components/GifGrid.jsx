import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({category}) => {
    const {images, isLoading} = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
            
            {
                isLoading && (<div>Cargando...</div>)
            }

            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem 
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}