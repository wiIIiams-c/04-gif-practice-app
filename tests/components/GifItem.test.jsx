import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas en <GifItem />', () => {
    const title = 'Bocchi'
    const url = 'https://bocchi-the-rock.cl/bocchi.jpg'

    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url}/>)
        //expect(screen.getByRole('img').src).toBe(url);
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
    })
})