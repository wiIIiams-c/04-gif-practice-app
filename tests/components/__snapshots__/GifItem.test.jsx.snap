// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Bocchi"
      src="https://bocchi-the-rock.cl/bocchi.jpg"
    />
    <p>
      Bocchi
    </p>
  </div>
</div>
`;
